utils/Azure/acrTools.ts

```diff
@@ -214,27 +215,19 @@ export async function getAzureImages(element: Repository): Promise<AzureImage[]>
  * @param registry : the registry to get login credentials for
  * @param context : if command is invoked through a right click on an AzureRepositoryNode. This context has a password and username
  */
-export async function loginCredentials(subscription: SubscriptionModels.Subscription, registry: Registry, context?: AzureImageNode | AzureRepositoryNode): Promise<{ password: string, username: string }> {
-    let node: AzureImageNode | AzureRepositoryNode;
-    if (context) {
-        node = context;
-    }
-    let username: string;
-    let password: string;
 
-    const client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
-    const resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+export async function loginCredentialsRefreshToken(subscription: SubscriptionModels.Subscription, registry: Registry, context?: AzureImageNode | AzureRepositoryNode): Promise<{ password: string, username: string }> {
+    let creds = await getRegistryTokens(registry);
+    let password = creds.refreshToken;
+    let username = '00000000-0000-0000-0000-000000000000';
+    return { password, username };
+}
 
-    if (registry.adminUserEnabled) {
-        let creds = await client.registries.listCredentials(resourceGroup, registry.name);
-        password = creds.passwords[0].value;
-        username = creds.username;
-    } else {
-        //grab the access token to be used as a password, and a generic username
-        let creds = await getRegistryTokens(registry);
-        password = creds.accessToken;
-        username = NULL_GUID;
-    }
+/* Used in delete Azure image and is pending edits */
+export async function loginCredentialsAccessToken(subscription: SubscriptionModels.Subscription, registry: Registry, context?: AzureImageNode | AzureRepositoryNode): Promise<{ password: string, username: string }> {
```

Comment:
>loginCredentialsAccessToken [](start = 22, length = 27)

 I recommend to return a credential object which contains both access and refresh token. But I will read more how the code use the two apis.